src/ui/ViewModelBase.cpp

```diff
@@ -0,0 +1,142 @@
+#include "ViewModelBase.hh"
+
+namespace ra {
+namespace ui {
+
+void ViewModelBase::SetValue(const BoolModelProperty& pProperty, bool bValue)
+{
+    IntModelProperty::ValueMap::const_iterator iter = m_mIntValues.find(pProperty.GetKey());
+
+    if (bValue == pProperty.GetDefaultValue())
+    {
+        if (iter == m_mIntValues.end())
+            return;
+
+        m_mIntValues.erase(iter);
+    }
+    else
+    {
+        if (iter != m_mIntValues.end())
+            return;
+
+        m_mIntValues[pProperty.GetKey()] = static_cast<int>(bValue);
+    }
+
+#ifdef _DEBUG
+    m_mDebugValues[pProperty.GetPropertyName()] = bValue ? L"true" : L"false";
```

Comment:
The raw hash maps the unique identifier to the value. The debug hash maps the property name to a string representation of the value for viewing in the debugger - it should never be read in the code.